File: src/plugin/object/NexEntranceObjectPlugin.java
package plugin.object;

import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.entity.object.WorldObject;
import org.redrune.game.plugin.type.ObjectPlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 9/1/2017
 */
public class NexEntranceObjectPlugin extends ObjectPlugin {
	
	@Override
	public boolean handle(Player player, WorldObject object, String option) {
		switch (option) {
			case "Climb-over":
				player.getDialogueManager().startDialogue("NexEntrance");
				return true;
		}
		return false;
	}
	
	@Override
	public void register() {
		register(57225, "Climb-over");
	}
}

File: src/main/java/plugin/object/NexEntranceObjectPlugin.kt
package plugin.`object`

import game.content.plugin.type.ObjectPlugin
import game.entity.`object`.WorldObject
import game.entity.actor.player.Player

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 9/1/2017
 */
class NexEntranceObjectPlugin : ObjectPlugin {
    override fun handle(player: Player, `object`: WorldObject, option: String): Boolean {
        when (option) {
            "Climb-over" -> {
                player.dialogueManager.startDialogue("NexEntrance")
                return true
            }
        }
        return false
    }

    override fun register() {
        registerObject(57225, "Climb-over")
    }
}